.eslintrc.json

```diff
@@ -1,6 +1,12 @@
 {
   "extends": "react-app",
   "rules": {
-    "curly": ["error"]
+    "curly": ["error"],
+    "valid-typeof": ["error"],
+    "camelcase": "error",
+    "id-length": ["error", { "min": 3, "exceptions": ["_","$","a","b","d","e","i","j","k","x","y","id","el","pi","PI"] }],
```

Comment:
I don't recall this rule being mentioned in our previous discussion - what is the premise or discussion behind adding this rule?

Reply:
hi Susanna, the rule in style-guide is not to use single-letter variable names but more meaningful names and hence we put up this rule. However, we specified the minimum length to be 3 

 But there are a lot of exception to this rule as double letter words have been used a lot in the graphing algorithm (ax,bx) so in certain places I have put up the eslint ignore rule and for some I have added them to exceptions.

Reply:
yes I do see the rule is for restricting variable naming, my question relates more to the context as to why this rule was added - is this something added on a further discussion of the JS lint rules that was advised to be added in via a separate discussion after our team discussion on the JS lint rules last Monday?

Furthermore, with regards to the exceptions, I wonder what is the significance to include single letters such as `"a","b","d","e","i","j","k"` in the rule? While I understand the exception of  "x", "y", "id", etc, specifying those other letters seems to defeat the idea behind setting this rule, which is for us to create meaningful and readable variable names. 

For ESlint rules, from how I see it, should relate to general best practice that can be applied universally. If there is so many possible exceptions that can apply to this rule, then it might be worth revisiting whether this is something that we should specify as a linter rule.

Reply:
I agree. Even I am slightly confused as to what to include in the exception. I can provide reasoning behind the above but I believe we can make this rules better. Let's set up sometime with Lim/Liam to go again through it and align and agree together. I will send a MR

Reply:
I think this is pretty well covered by this section of style guide here with the general reasoning behind it:

https://github.com/quantumblacklabs/kedro-viz/blob/main/STYLE_GUIDE.md#variable-naming

This lint rule of avoiding 1 or 2 letter names is to try to encourage clearer naming. 

There are fair exceptions for the cases where the naming is well established by standard practices - so `i` and `j` for loops are standard in JS, `x` and `y` for position are standard in mathematics (ideally these would be written into the lint config, but don't think comments are possible there)